airbyte-integrations/connectors/source-google-ads/source_google_ads/source.py

```diff
@@ -47,13 +60,33 @@
 )
 from .utils import GAQL, logger, traced_exception
 
+_MAX_CONCURRENCY = 20
+_DEFAULT_CONCURRENCY = 10
+
 
-class SourceGoogleAds(AbstractSource):
+class SourceGoogleAds(ConcurrentSourceAdapter):
+    message_repository = InMemoryMessageRepository(logger.level)
     # Skip exceptions on missing streams
     raise_exception_on_missing_stream = False
 
+    def __init__(self, catalog: Optional[ConfiguredAirbyteCatalog] = None,
+                 config: Optional[Mapping[str, Any]] = None,
+                 state: TState = None, **kwargs):
+        if config:
+            concurrency_level = min(config.get("num_workers", _DEFAULT_CONCURRENCY), _MAX_CONCURRENCY)
+        else:
+            concurrency_level = _DEFAULT_CONCURRENCY
+        logger.info(f"Using concurrent cdk with concurrency level {concurrency_level}")
+        concurrent_source = ConcurrentSource.create(
+            concurrency_level, concurrency_level // 2, logger, self._slice_logger, self.message_repository
```

Comment:
Since we do `concurrency_level // 2` here, the min concurrency in the config needs to be 2.

Reply:
Fixed